invenio_files_rest/__init__.py

```diff
@@ -6,7 +6,744 @@
+>>> app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite://'
+>>> app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+>>> app.config['FILES_REST_PERMISSION_FACTORY'] = allow_all
+
+
+Now let's initialize all required Invenio extensions:
+
+>>> import shutil
+>>> from os import makedirs
+>>> from os.path import dirname, exists
+>>> from pprint import pprint
+>>> import json
+
+>>> from flask_babelex import Babel
+>>> from flask_menu import Menu
+>>> from invenio_db import InvenioDB, db
+>>> from invenio_rest import InvenioREST
+>>> from invenio_admin import InvenioAdmin
+>>> from invenio_accounts import InvenioAccounts
+>>> from invenio_access import InvenioAccess
+>>> from invenio_accounts.views import blueprint as accounts_blueprint
+>>> from invenio_celery import InvenioCelery
+>>> from invenio_files_rest import InvenioFilesREST
+>>> from invenio_files_rest.views import blueprint
+>>> from invenio_files_rest.models import Location
+
+>>> Babel(app)
+>>> Menu(app)
+>>> InvenioDB(app)
+>>> InvenioREST(app)
+>>> InvenioAdmin(app)
+>>> InvenioAccounts(app)
+>>> InvenioAccess(app)
+
+Finally, let's initialize InvenioFilesREST, register the blueprints
+and push a Flask application context:
+
+>>> InvenioFilesREST(app)
+
+>>> app.register_blueprint(accounts_blueprint)
+>>> app.register_blueprint(blueprint)
+
+>>> app.app_context().push()
+
+Let's create the database and tables, using an in-memory SQLite database:
+
+>>> db.create_all()
+
+>>> srcroot = dirname(dirname(__file__))
+>>> d = app.config['DATADIR']
+>>> if exists(d):
+>>>     shutil.rmtree(d)
+>>> makedirs(d)
+
+There's also a need to create a location for the buckets:
+
+>>> loc = Location(name='local', uri=d, default=True)
+>>> db.session.add(loc)
+>>> db.session.commit()
+
+Now let's create a bucket:
+
+>>> res = app.test_client().post('/files')
+
+And see the response containing the id of the bucket:
+
+>>> pprint(json.loads(res.get_data().decode("utf-8")))
+
+
+REST API Overview
+-----------------
+
+This part of the documentation will show you how to get started in using the
+REST API of Invenio-Files-REST.
+
+.. note::
+    Note that the api is exposed from :code:`/files/` and is registered as a
+    blueprint_api which means in our wrapper application will be prefixed with
+    :code:`/api/`. That means that our api will be exposed in our application
+    under :code:`/api/files/`.
+
+
+About the API
+-------------
+
+The REST API allows you to create buckets and perform CRUD operations on files.
+You can use query parameters in order to perform these operations.
+
+
+Available methods and endpoints
+-------------------------------
+
+By default, the URL prefix for the REST API is under /files.
+
+HEAD
+^^^^
+
+Check if bucket exists, returning either a 200 or a 404:
+
+.. code-block:: console
+
+    HEAD /files/<bucket_id>
+
+GET
+^^^
+
+Return list of files in bucket:
+
+.. code-block:: console
+
+    GET /files/<bucket_id>/
+
+Returns list of file versions:
```

Comment:
The form of the verb is inconsistent- I would stay with "Return" also for the next 2 occurrences.